src/compute/src/render/top_k.rs

```diff
@@ -433,12 +460,14 @@ where
                 }
             });
 
-            // We arrange the inputs ourself to force it into a leaner structure because we know we
-            // won't care about values.
-            let consolidated = collection
-                .consolidate_named::<RowKeySpine<_, _, _>>("Consolidated MonotonicTop1 input");
+            // Consolidate if required to do so.
+            if must_consolidate {
```

Comment:
The repeated pattern of 

```rust
if must_consolidate {
   collection = collection.consolidate_named("name");
}
```

makes me thing that this can be abstracted away in a method (possibly in an extension trait) so that we can just write the more concise:

```rust
let mut collection = collection
    .map({
        let mut datum_vec = mz_repr::DatumVec::new();
        move |row| {
            // ...
        }
    }).consolidate_named_if(must_consolidate, "Consolidated MonotonicTop1 input");
```

instead of relying on `mut collection` bindings and an `if` block.

Reply:
I like this suggestion, thanks! I added the new method to `CollectionExt` (see f3466e8d4aa3d36e061a5e7d895f0a1e1931ea73).